src/pages/pin.tsx

```diff
@@ -1,44 +1,55 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { PinInput, HStack, PinInputField, Center } from '@chakra-ui/react';
 import { useRouter } from 'next/router';
 import { MainPageWrapper } from '../components/wrappers/main-page-wrapper';
 import { trpc } from '../utils/trcp';
 import { QuizData, useQuizDataStore } from '../state/quiz-data.state';
+import { PIN_LENGTH } from '../../constants';
 
 const Pin = () => {
   const [pin, setPin] = useState('');
   const setQuizData = useQuizDataStore((state) => state.setQuizData);
 
   const router = useRouter();
 
-  trpc.team.getQuizByPin.useQuery(
-    { pin },
-    {
-      enabled: pin.length === 4,
-      onSuccess: (quizData) => onGettingQuizDataSuccessfully(quizData),
-    },
-  );
-
   function onGettingQuizDataSuccessfully(quizData: QuizData) {
     setQuizData(quizData);
 
-    router.push(`/${quizData.id}`);
+    router.push({
```

Comment:
why this is changed?

Reply:
Both work. I prefer the changed syntax because, to me, thats a little more reusable and also readable. But I guess its your call.
Also the quizId: 1 was just as a mock. I will revert that back to quizData.id